ticdc: add description about gc-ttl (#6835)
pingcap/docs: ticdc/troubleshoot-ticdc.md

When the replication task is unavailable or interrupted, this feature ensures that the data to be consumed by TiCDC is retained in TiKV without being cleaned by GC.

When starting TiCDC server, you can specify the TTL of GC safepoint by `gc-ttl`. You can also [use TiUP to modify `gc-ttl`](/ticdc/manage-ticdc.md#modify-ticdc-configuration-using-tiup). The default value is 24 hours. In TiCDC, this value means:

- The maximum time the GC safepoint will be retained at the PD when the TiCDC service is stopped.
- The maximum time a replication task can be suspended when interrupted or manually stopped. If a replication task is suspended longer than the value set by `gc-ttl`, the replication task will enter the `failed` state, cannot be resumed, and will not continue to affect the progress of the GC safepoint.

The second behavior above is introduced in TiCDC v4.0.13 and later versions. The purpose is to prevent a replication task in TiCDC from suspending for too long, causing the GC safepoint of the upstream TiKV cluster not to continue for a long time and retaining too many outdated data versions, thus affecting the performance of the upstream cluster.

> > **Note**
> > In some scenarios, such as when using TiCDC for incremental replication after full replication with Dumpling/BR, the default 24 hours of `gc-ttl` may not be sufficient. You need to specify an appropriate value for `gc-ttl` when you start the TiCDC server.

Reviewer: ```suggestion
> **Note:**
>
> In some scenarios, for example, when you use TiCDC for incremental replication after full replication with Dumpling/BR, the default 24 hours of `gc-ttl` may not be sufficient. You need to specify an appropriate value for `gc-ttl` when you start the TiCDC server.
```
这个对应的中文也少一行，还请@hfxsd 再提个 PR 改一下中文吧~

## What is the complete behavior of TiCDC garbage collection (GC) safepoint?

If a replication task starts after the TiCDC service starts, the TiCDC owner updates the PD service GC safepoint with the smallest value of `checkpoint-ts` among all replication tasks. The service GC safepoint ensures that TiCDC does not delete data generated at that time and after that time. If the replication task is interrupted, or manually stopped, the `checkpoint-ts` of this task does not change. Meanwhile, PD's corresponding service GC safepoint is not updated either.

> If the replication task is suspended longer than the time specified by `gc-ttl`, the replication task enters the `failed` state and cannot be resumed. The PD corresponding service GC safepoint will continue.

Reviewer: ```suggestion
If the replication task is suspended longer than the time specified by `gc-ttl`, the replication task enters the `failed` status and cannot be resumed. The PD corresponding service GC safepoint will continue.
```
